### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by mariomoore**

### CODE REVIEW

File: module2/homework/nwdNww/nwdNww.hpp [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
+    return higher;
 }
 
-int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+int NWW(int lhs, int rhs)
+{
+    int nwd = NWD(lhs, rhs);
+    if (nwd)
```
Comment (anraf1001) [inline module2/homework/nwdNww/nwdNww.hpp:23]: Implicit casting to boolean. Comparing to 0 is much clearer
```suggestion
    if (nwd != 0)
```

File: module2/homework/nwdNww/nwdNww.hpp [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+    {
+        return std::abs(lhs * rhs) / nwd;
```
Comment (Rosein) [inline module2/homework/nwdNww/nwdNww.hpp:25]: I'm copying piezol's comment from different PR:
There are trivial cases that your code won't work for, ie NWW(1e8, 1e8).
That is due to lhs*rhs overflowing int.
You can fix it by casting lhs to unsigned long long in the multiplication.

File: module2/homework/generateSequence/generateSequence.hpp [hunk 3]
```diff
@@ -1,7 +1,20 @@
...
+{
+    if (count > 0)
+    {
+        std::vector<int> sequence(count);
+        int val2add = step;
+        for (auto& i : sequence)
+        {
+            i = val2add;
+            val2add += step;
+        }
```
Comment (nauka-programowania-MA) [inline module2/homework/generateSequence/generateSequence.hpp:14]: you can avoid val2add 
just use sequence * -> when you use for loop (int i.....)
or sequence[i] = sequence[i-1] + step -> but remember to assign value to sequence[0] and iterator form 1 to count

**Comment by ziobron** [comment]
{ "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635], "points": [2, 4, 2, 6, 2, 5, 2, 5] }

**Pull request closed without merging**